crepecake/src/main/java/com/google/cloud/tools/crepecake/blob/BlobDescriptor.java

```diff
@@ -46,14 +47,24 @@ public BlobDescriptor(long size, DescriptorDigest digest) {
     this.digest = digest;
   }
 
+  /** Initialize with just size. */
```

Comment:
when would one partially initialize a blob?

Reply:
This is a BlobDescriptor that is used in FileBlob and InputStreamBlob, where the bytes are counted while piping but no hash is calculated. It is also used in EmptyBlob and especially StringBlob, where the digest doesn't apply, like an image manifest.

Reply:
It's hard for me to tell, but do you think this is an indication that we could have overgeneralized our classes?

Reply:
Yeah, that's definitely possible. I'll file an issue to think more about this. BlobDescriptor was intended to represent the content descriptors in an image manifest, which have size and digest. It was put into one class to return either or both from Blob#writeTo so that the digest and size of the blob can be captured upon write.